diff --git a/board/compulab/ucm-imx93/ucm-imx93.c b/board/compulab/ucm-imx93/ucm-imx93.c
index 3695ca0f460..3fdb24157bd 100644
--- a/board/compulab/ucm-imx93/ucm-imx93.c
+++ b/board/compulab/ucm-imx93/ucm-imx93.c
@@ -180,10 +180,19 @@ int board_late_init(void)
 	env_set("board_name", "UCM-iMX93");
 	env_set("board_rev", "iMX93");
 #endif
-	board_get_mac_from_eeprom(0);
-	board_get_mac_from_eeprom(1);
+//	board_get_mac_from_eeprom(0);
+//	board_get_mac_from_eeprom(1);
 
 	printf("%s [ 0x%llx ]\n",__func__,gd->ram_size);
+        printf("\n");
+        printf("+----------------------------------------------+\n");
+        printf("|   Target Board: UCM - i.MX93 EXP-Satellite   |\n");
+        printf("|          BEE-1006 - PAY-EXP V1.1.0           |\n");
+        printf("|                  Booting-up!                 |\n");
+        printf("+----------------------------------------------+\n");
+        printf("\n");
+        printf("U-Boot Version: Rev 1.0.1\n");
+        printf("Kernel Version: Rev 1.0.1\n");
 	return 0;
 }
 
diff --git a/drivers/gpio/pca953x_gpio.c b/drivers/gpio/pca953x_gpio.c
index c81d2963dc8..6352220c7c8 100644
--- a/drivers/gpio/pca953x_gpio.c
+++ b/drivers/gpio/pca953x_gpio.c
@@ -392,7 +392,7 @@ static const struct udevice_id pca953x_ids[] = {
 	{ .compatible = "nxp,pca9538", .data = OF_953X(8, PCA_INT), },
 	{ .compatible = "nxp,pca9539", .data = OF_953X(16, PCA_INT), },
 	{ .compatible = "nxp,pca9554", .data = OF_953X(8, PCA_INT), },
-	{ .compatible = "nxp,pca9555", .data = OF_953X(16, PCA_INT), },
+	{ .compatible = "nxp,pca9555-DISABLED", .data = OF_953X(16, PCA_INT), },
 	{ .compatible = "nxp,pca9556", .data = OF_953X(8, 0), },
 	{ .compatible = "nxp,pca9557", .data = OF_953X(8, 0), },
 	{ .compatible = "nxp,pca9574", .data = OF_957X(8, PCA_INT), },
