diff --git a/include/configs/compulab-imx93.h b/include/configs/compulab-imx93.h
index 1b2b98f8646..d7cebfda009 100644
--- a/include/configs/compulab-imx93.h
+++ b/include/configs/compulab-imx93.h
@@ -84,7 +84,7 @@
 	"mmcpart=1\0" \
 	"mmcroot=" CONFIG_MMCROOT " rootwait rw\0" \
 	"mmcautodetect=yes\0" \
-	"mmcargs=setenv bootargs ${jh_clk} console=${console} root=${mmcroot}\0 " \
+	"mmcargs=setenv bootargs ${jh_clk} clk_ignore_unused clk-imx93.mcore_booted=1 console=${console} root=${mmcroot}\0 " \
 	"loadbootscript=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
